core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -241,13 +247,85 @@ class QualificationAppInfo(
     stages.map { stageId =>
       val stageTaskTime = stageIdToTaskEndSum.get(stageId)
         .map(_.totalTaskDuration).getOrElse(0L)
+      val numTransitions = penalizeTransitions match {
+        case true => stageIdToGpuCpuTransitions.getOrElse(stageId, 0)
+        case false => 0
+      }
+      val transitionsTime = numTransitions match {
+        case 0 => 0L // no transitions
+        case gpuCpuTransitions =>
+          // Duration to transfer data from GPU to CPU and vice versa.
+          // Assuming it's a PCI-E Gen3, but also assuming that some of the result could be
+          // spilled to disk.
+          // Duration in Spark metrics is in milliseconds and CPU-GPU transfer rate is in bytes/sec.
+          // So we need to convert the transitions time to milliseconds.
+          val totalBytesRead = {
+            stageIdToTaskEndSum.get(stageId).map(_.totalbytesRead).getOrElse(0L)
+          }
+          if (totalBytesRead > 0) {
+            val transitionTime = (totalBytesRead /
+              QualificationAppInfo.CPU_GPU_TRANSFER_RATE.toDouble) * gpuCpuTransitions
+            (transitionTime * 1000).toLong // convert to milliseconds
+          } else {
+            0L
+          }
+
+        case _ => 0L
+      }
+      val finalEachStageUnsupported = if (transitionsTime != 0) {
+        // Add 20% penalty for unsupported duration if there are transitions.
+        (eachStageUnsupported * 0.2 + eachStageUnsupported).toLong
```

Comment:
what is we make ths 0.5?    Does that make the customer logs and then our fake NDS queries numbers closer or does that penalize to much?    Either way we should also put a sentence that says this number was randomly picked because it matched roughly what we saw on the experiments with customer/nds event logs

Reply:
After updating it to 0.5, it makes the customer logs closer to observed speedups. On NDS queries, it penalizes couple of queries: 
query 14 - estimated: 1.58, observed: 2.17
query67 - estimated: 1.45, observed: 1.76.

Added comment